shared/src/commonMain/kotlin/org/hyperskill/app/paywall/domain/analytic/PaywallClickedBuySubscriptionHyperskillAnalyticEvent.kt

```diff
@@ -15,15 +16,26 @@ import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticTar
  *     "route": "/paywall",
  *     "action": "click",
  *     "part": "main",
- *     "target": "buy_subscription"
+ *     "target": "buy_subscription",
+ *     "context":
+ *     {
+ *       "source": "login"
+ *     }
  * }
  * ```
  *
  * @see HyperskillAnalyticEvent
  */
-object PaywallClickedBuySubscriptionHyperskillAnalyticEvent : HyperskillAnalyticEvent(
+class PaywallClickedBuySubscriptionHyperskillAnalyticEvent(
+    private val paywallTransitionSource: PaywallTransitionSource
+) : HyperskillAnalyticEvent(
     HyperskillAnalyticRoute.Paywall,
     HyperskillAnalyticAction.CLICK,
     HyperskillAnalyticPart.MAIN,
     HyperskillAnalyticTarget.BUY_SUBSCRIPTION
-)
\ No newline at end of file
+) {
+    override val params: Map<String, Any>
+        get() = super.params + mapOf(
+            PaywallAnalyticKeys.PAYWALL_TRANSITION_SOURCE to paywallTransitionSource.analyticName
+        )
```

Comment:
To put value to the `context` you need to use a different syntax:
```kotlin
override val params: Map<String, Any>
    get() = super.params +
        mapOf(
            PARAM_CONTEXT to mapOf(
                PaywallAnalyticKeys.PAYWALL_TRANSITION_SOURCE to paywallTransitionSource.analyticName
            )
        )
```

Reply:
please also update all paywall analytic events `params` computed properties

Reply:
@ivan-magda 
I suggest consider a special property, like `contextParams: Map<String, Any>`, to easier add context params